File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int[] answer = new int[2];

        int numer3 = denom1 * denom2;
        int denom3 = numer1 * denom2 + numer2 * denom1;
        int max = 1;

        for (int i = 1; i <= numer3 && i <= denom3; i++) {
            if (numer3 % i == 0 && denom3 % i == 0) {
                max = i;
            }
        }
        answer[0] = denom3 / max;
        answer[1] = numer3 / max;

        return answer;
    }
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.kt
class Solution {
    fun gcd(a: Int, b: Int): Int = if (b != 0) gcd(b, a % b) else a

    fun solution(numer1: Int, denom1: Int, numer2: Int, denom2: Int): IntArray {
        var answer: IntArray = intArrayOf()

        val number3 = numer1 * denom2 + numer2 * denom1
        val denon3 = denom1 * denom2

        return intArrayOf(number3 / gcd(number3, denon3), denon3 / gcd(number3, denon3))
    }
}
